cmdLine/src/main/scala/topgun/cmdline/ClassLoaderInfo.scala

```diff
@@ -14,18 +15,22 @@ class ClassLoaderInfo() {
   }
 
   def buildClassInfo(className: String, classLoader: ClassLoader): ClassInfo = {
-    val resourceName = className.replace(".", "/") + ".class"
-    val iStream = classLoader.getResourceAsStream(resourceName)
-    val reader = new ClassReader(iStream)
-    val classNode = new ClassNode()
-    // specify no parsing options.
-    val parsingFlagNone = 0;
-    reader.accept(classNode, parsingFlagNone)
-    iStream.close()
-    val classNameFromReader = reader.getClassName
-    val sourceFile = classNode.sourceFile
-    val superClass = reader.getSuperName
-    val interfaces:List[String] = reader.getInterfaces.toList
-    ClassInfo(classNameFromReader, superClass, sourceFile, interfaces)
+    try {
+      val resourceName = className.replace(".", "/") + ".class"
+      val iStream = classLoader.getResourceAsStream(resourceName)
+      val reader = new ClassReader(iStream)
+      val classNode = new ClassNode()
+      // specify no parsing options.
+      val parsingFlagNone = 0;
+      reader.accept(classNode, parsingFlagNone)
+      iStream.close()
+      val classNameFromReader = reader.getClassName
+      val sourceFile = classNode.sourceFile
+      val superClass = reader.getSuperName
+      val interfaces: List[String] = reader.getInterfaces.toList
+      ClassInfo(classNameFromReader, superClass, sourceFile, interfaces)
+    }catch {
+      case e: IOException => ClassInfo("className not found","super not found","sourceFile not found",List())
```

Comment:
too general a catch

Reply:
The ASM ClassReader constructor throws that IOException and the message in there is class not found. 
public ClassReader(final String className) throws IOException

Reply:
and in that case, what is the InputStream?

Reply:
fixed. 
InputStream was null

Reply:
You marked this as resolved, but the code is unchanged

Reply:
It has changed. I placed a check to see if the input stream is null before continuing to the ClassReader.